File: WidgetsDemo/LayoutDemo/app/src/main/java/com/example/layoutdemo/MainActivity.java
package com.example.layoutdemo;

import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import static android.content.res.Configuration.ORIENTATION_LANDSCAPE;
import static android.content.res.Configuration.ORIENTATION_PORTRAIT;
import static com.example.layoutdemo.R.id.button_port;
import static com.example.layoutdemo.R.id.hello;
import static com.example.layoutdemo.R.id.left_button_land;
import static com.example.layoutdemo.R.id.right_button_land;
import static com.example.layoutdemo.R.layout.activity_main;

public class MainActivity extends AppCompatActivity {

	TextView textView;
	Button centerButtonPort;
	Button leftButtonLand, rightButtonLand;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(activity_main);

		textView = findViewById(hello);
		centerButtonPort = findViewById(button_port);
		leftButtonLand = findViewById(left_button_land);
		rightButtonLand = findViewById(right_button_land);
		textView.setText("HI");
		if (getResources().getConfiguration().orientation == ORIENTATION_PORTRAIT)
			centerButtonPort.setText("My Button");
		else if (getResources().getConfiguration().orientation == ORIENTATION_LANDSCAPE) {
			leftButtonLand.setText("Left");
			rightButtonLand.setText("Right");
		}
	}
}

File: BottomNavTesting/app/src/main/java/com/example/cookie_clicker/MainActivity.kt
package com.example.cookie_clicker

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.example.cookie_clicker.R.id.*
import com.example.cookie_clicker.R.layout.activity_main
import com.google.android.material.bottomnavigation.BottomNavigationView
import java.util.concurrent.atomic.*


class MainActivity : AppCompatActivity() {
	companion object {
//		val storage = Firebase.storage.reference
//		var jsonObject = JSONObject()
		var cookie_quantity: AtomicReference<Double> = AtomicReference(0.0)
		var cps: AtomicReference<Double> = AtomicReference(0.0)
		var isCpSOccurring = AtomicBoolean()
	}

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(activity_main)
//		val localFile = createTempFile(
//			"cookie_clicker",
//			"json",
//			File("src\\main\\java\\com\\example\\cookie_clicker")
//		)
//		storage.child("/cookie_clicker.json").getFile(localFile)
//		val json: String
//		val inputStream = applicationContext.assets.open("file_name.json")
//		val buffer = ByteArray(inputStream.available())
//		inputStream.read(buffer)
//		inputStream.close()
//		json = String(buffer, defaultCharset())
//		jsonObject = JSONObject(json)
//		localFile.delete()
		val navController = findNavController(nav_host_fragment)
		setupActionBarWithNavController(
			navController,
			AppBarConfiguration(setOf(navigation_cookie, navigation_buildings))
		)
//		cookie_quantity.set(jsonObject.getDouble("cookies"))
//		val jsonArray = jsonObject.getJSONArray("building_quantities")
//		(0..jsonArray.length()).forEach()
		findViewById<BottomNavigationView>(nav_view).setupWithNavController(navController)
//		val newLooper = Looper.
		/*val handler = Handler(mainLooper)
		val runnableCode: Runnable = object : Thread() {
			override fun run() {
				isCpSOccurring.getAndSet(true)
				cookie_quantity.getAndUpdate { it + cps.get() }
				val temp = truncate(cookie_quantity.get()).toString()
				CookieFragment.cookiesTextView.get().text = temp.substring(0, temp.indexOf('.'))
				isCpSOccurring.getAndSet(false)
				handler.postDelayed(this, 1000)
			}
		}
		handler.post(runnableCode)*/
	}

//	override fun onStop() {
//		super.onStop()
//		val jsonToPush = JSONObject()
//		jsonToPush.put("cookies", cookie_quantity.get())
//		jsonToPush.put("cps", cps.get())
//		val list = arrayListOf<JSONObject>()
//		(0..17).forEach { i ->
//			list += JSONObject().put(
//				BuildingFragment.buyButtons[i].name.toString().replace(" ".toRegex(), "_")
//					.toLowerCase(ROOT) + "_quantity", BuildingFragment.buildingQuantities[i]
//			)
//		}
//		jsonToPush.put("building_quantities", JSONArray(list))
//		val localFile = createTempFile(
//			"cookie_clicker",
//			"json",
//			File("src\\main\\java\\com\\example\\cookie_clicker")
//		)
//		FileWriter(localFile).append(jsonToPush.toString())
//		storage.putFile(Uri.fromFile(localFile))
//	}
}